### PR TITLE
Custom mustache delimiters

### PR DESCRIPTION
**Pull request by alabiaga**
Fixes: #12994 

- [x] add support for custom delimiters in script tag defined templates.
- [x] add support for custom delimiters in template tag defined templates.

Allow for defining custom delimiters via data-custom-delimiters attribute.

e.g.

    <amp-list ..>
	  <template type="amp-mustache" data-custom-delimiters="<%,%>">
         ...
      </template>
	</amp-list>

    or

    <amp-list ..>
	  <script type="text/plain" template="amp-mustache"
          data-custom-delimiters="<%,%>">
         ...
      </script>
	</amp-list>

### CODE REVIEW

**Comment by alabiaga** [comment]
cc\ @choumx The changes required to support custom delimiters in script defined templates as we discussed offline. I haven't worked on supporting template tags yet but opening up for review just for consensus on defining the delimiters on the script or template tag.

**Comment by alabiaga** [comment]
cc\ @morsssss

File: extensions/amp-mustache/0.1/amp-mustache.js [hunk 1]
```diff
@@ -61,30 +61,66 @@ export class AmpMustache extends AMP.BaseTemplate {
...
+   *     so that default mustache delimiters are used.
+   * @private
+   */
+  getDelimiters_() {
+    const CUSTOM_DELIMITERS_ATTR = 'data-custom-delimiters';
+    let delimiters = null;
+    const {element} = this;
+    if (element.hasAttribute(CUSTOM_DELIMITERS_ATTR)) {
+      const delimitersStr = element.getAttribute(CUSTOM_DELIMITERS_ATTR);
+      devAssert(delimitersStr.split(',').length == 2,
```
Comment (jridgewell) [inline extensions/amp-mustache/0.1/amp-mustache.js]: `userAssert`

**Comment by alabiaga** [comment]
apologies closing this one, in favor of https://github.com/ampproject/amphtml/pull/21460. Trying to collapse all the commits into one as it's been an issue with all my branches..and something funky happened. Thanks

**Pull request closed without merging**